src/dialect/postgresql.rs

```diff
@@ -138,6 +138,14 @@ impl Dialect for PostgreSqlDialect {
     fn parse_statement(&self, parser: &mut Parser) -> Option<Result<Statement, ParserError>> {
         if parser.parse_keyword(Keyword::COMMENT) {
             Some(parse_comment(parser))
+        } else if parser.parse_keyword(Keyword::CREATE) {
+            match parse_create(parser) {
+                Some(result) => Some(result),
+                None => {
+                    parser.prev_token(); // unconsume the CREATE if we didn't end up parsing anything
+                    None
+                }
```

Comment:
it might be possible to simplify this if we call `prev_token()` before calling `parse_create`? thinking since the latter already returns an Option representing the same result

Reply:
Yes, that can make it much simpler. I also tried using peek to avoid doing "consume, unconsume, consume" but it looks like there's no nice `peek_keyword()` type function and so it ended up looking like:
```rust
if matches!(parser.peek_token().token, Token::Word(Word { keyword: Keyword::CREATE, .. })) {
    parse_create(parser)
}
```
That's a bit of a mouthful and I ended up just going with your suggestion:
```rust
if parser.parse_keyword(Keyword::CREATE) {
    parser.prev_token(); // unconsume the CREATE in case we don't end up parsing anything
    parse_create(parser)
}
```